M3 architecture

IvanSkripov opened the pull request:
**Описание**

Описан проект - Галерия изображений. Главные моменты - минималистичный интерфейс, отсутствие «подсунутого» контента, использование ИИ для аннотации изображений Описание MVP дано по набору функций в README.md (раздел Основной функционал) Обновлен docker-compose + первичные заметки по DevOps

**Скриншоты** 
OpenSearch с выводом логов NGINX (имитация приложения) + Envoy (для удобства анализа работы с keyclock)

_NGINX (вывод распарсен)_
<img width="1530" height="565" alt="Nginx" src="https://github.com/user-attachments/assets/40faf637-2907-4cb0-a307-badf3fd6bab7" />

_ENVOY (вывод в поле log)_
<img width="1510" height="701" alt="envoy" src="https://github.com/user-attachments/assets/50e5d5bf-7e03-4d38-9fc4-927e2cf4c222" />

IvanSkripov committed:
Arch. Plugins (v.1)

IvanSkripov committed:
Add Minio container and envoy logs

IvanSkripov committed:
Add Minio container and envoy logs

IvanSkripov committed:
Add DevOps notes

IvanSkripov committed:
Add API Descriptions

IvanSkripov committed:
Add minor changes

IvanSkripov committed:
Add .gitignore

IvanSkripov committed:
Add envoy logs start

Commit:
Add FluentBit Nginx parser

IvanSkripov committed:
Update Documentation

svok commented on docs/01-biz/01-target-audience.md:
```diff
@@ -0,0 +1,60 @@
+## Гипотетические портреты пользователей сервиса
+| Роль на Площадке                                                         | Описание                                                                         | Ключевые Демографические/Профессиональные Характеристики                                                                                                                                 | Главные Цели на Площадке                             ...
+|:-------------------------------------------------------------------------|:---------------------------------------------------------------------------------| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :--------------------------------------------------- ...
+| **Пользователь (Организация)**    | Наемный сотрудник, фрилансер (Мужчина, Женщина 50%-50%) <br> *Возраст: 24-45*    | Высшее техническое образование. Опыт работы 7+ лет. Принимает решения на основе ТТХ, ГОСТ/ТУ, сертификатов. Скептичен к маркетингу. Аналитичен.                                         | Найти **конкретный** компонент под ТЗ. Сравнить альтернативы по **техническим параметрам и ц ...
```
Все, кто пользуется компьютером - пользователи. Не стоит использовать бесполезные слова. Нужна конкретика. Это же не просто пользователь, а какой-нибудь художник или редактор

IvanSkripov replied:
Доработано

svok commented on docs/01-biz/01-target-audience.md:
```diff
@@ -0,0 +1,60 @@
+
+[//]: # (Образование: Высшее &#40;искусствовед&#41;)
+
+[//]: # (Профессия: Менеджер по арт-проектам)
```
Вот это и есть отличительная черта

svok commented on docs/03-architecture/02-api.md:
```diff
@@ -0,0 +1,79 @@
+
+## 2. Бизнес-функции (методы)
+**CRUDS для изображений**
+1. **`import`** - Импорт изображения
```
1. create все равно потребуется. Если бы у вас простой импорт был, да можно было бы обойтись. Но у вас здесь заголовок и описание, которые вводятся людьми.
2. Вам нужно проработать процесс. Либо это предварительная загрузка, а потом заполнение данных, либо это создание записи с последующей дозагрузкой содержимого. На текущий момент я не вижу как вы планируете создавать начальный объет. Импорт - это же просто файлы скачать откуда-то. А заполнять как?

svok commented on docs/03-architecture/02-api.md:
```diff
@@ -0,0 +1,79 @@
+2. **`read`** - Получение информации об изображении
+   - *Бизнес-правило*: Изображение принадлежит владельцу (id.owner == visitor)
+
+3. **`link`** - Получение public link на изображение
```
Мне кажется, read поностью решает все задачи link. Вам точно нужен этот метод? Имейте в виду, его создание и поддержка отнимают ресурсы.

svok commented on docs/03-architecture/02-api.md:
```diff
@@ -0,0 +1,79 @@
+   - *Бизнес-правило*: Новые теги добавляются в общий список
+
+6. **`delete`** - Удаление изображения
+   - *Бизнес-правило*: Только владелец может удалять изображение (id.owner == visitor)
```
Я себе вижу так, что здесь должен быть бы полный CRUDS для записи об изображении. Плюс еще upload и download для содержимого изображения. Import можно оставить на потом, потому как по нему пока не ясно как он должен работать и откуда брать данные.
Но это только мои мысли. Вы делайте как считаете правильным.

IvanSkripov replied:
Добавлен метод create. Import убран из рамок MVP

svok commented on docs/03-architecture/02-api.md:
```diff
@@ -0,0 +1,79 @@
+8. **`search`** - Поиск объявлений
+   - *Бизнес-правило*: Возвращает список найденных изображений по заданным критериям
+   - *Бизнес-правило*: Реализована пагинация
+   - *Бизнес-правило*: Изображения принадлежат владельцу (each(id.owner) == visitor)
```
Эти два метода - одно и то же. Опять же возвращаемся к вопросу поддержки лишних методов. crudS содержит в себе Search, этого достаточно.

IvanSkripov replied:
Метод list убран.

svok commented on docs/03-architecture/02-api.md:
```diff
@@ -0,0 +1,79 @@
+
+10. **`tags.list`** - Вывод списка всех тегов пользователя
+   - *Бизнес-правило*: Теги создаются автоматически при загрузке, аннотировании или редактировании объекта
+   - *Бизнес-правило*: Выводится список тегов для всех изображений
```
Давайте так. Если вы готовы поддерживать еще две сущности - labels, tags - то это должно быть два отдельных CRUDS для них. Думаю, что сейчас в рамках учебы это излишне.
Если вы не готовы добавлять еще две лишних сущности, то image.read вполне решает все вопросы с перечислением лейблов и тэгов. Опять вопрос поддержки лишних методов.

IvanSkripov replied:
Описаны границы применимости меток (метаданных) и тэгов. Поддержка CRUD в рамках MVP не требуется

svok commented on docs/03-architecture/02-api.md:
```diff
@@ -0,0 +1,79 @@
+| image       | Ссылка на изображение                                        | ../images/36df4909-37b7-4381-8a95-bb0ba387d8d9/5b5a5b43-e30e-4b2d-a39b-2a392321e4d4.jpg     |
+| preview     | Ссылка на preview                                            | ../previews/5b5a5b43-e30e-4b2d-a39b-2a392321e4d4.jpg                                             |
+
+**Элемент галереи (ImageItem)**
```
Я не понял что это за элемент и зачем он нужен. С учетом ограничений курса, я бы предложил его вообще не рассматривать.

svok commented on docs/03-architecture/arch-C4-Context.drawio.png:
Боюсь, вы недооцениваете сложность проекта. Задание на импорт - это отдельная сущность. Далее сервис импорта - это отдельный сервис и тоже немалый объем работ. Он должен идти отдельно от каталога изображений.
Вся обвязка, которую вы удалили из исходной архитектуры, она тоже нужна. Все роли тоже не с потолка взяты.

IvanSkripov replied:
Импорт в рамках MVP не рассматривается

svok commented on imgs/Wireframe.jpeg:
Здесь отлично, что вы представляете каталок. Только я вижу, что главная трудность у вас возникла с формой создания изображения (загрузка)

IvanSkripov replied:
Спасибо

svok reviewed (commented):
Иван, хорошая работа. Я оставил комментарии, рекомендую поработать с ними.

IvanSkripov committed:
Remove from API Элемент Галереи (ImageItem)

IvanSkripov committed:
Доработаны
1. Описание целевой аудитории
2. Убран метод list, его функционал заменяет search
3. Убран import и добавлен create. В рамках MVP import не будет рассматриваться
